### DISCUSSION TITLE
Reusable Workflows - how to reference own called repo from within a reusable workflow?

### DISCUSSION THREAD

**Discussion by HariSekhon** [community/community#18602] [Actions]
Related to #18601, is there any way to reference the repo of a reusable workflow within the reusable workflow steps itself, via something like the `env` or `github` contexts?

I've dumped all the contexts by calling a reusable workflow version which dumps all its contexts, but couldn't find anything appropriate:

https://github.com/HariSekhon/GitHub-Actions/blob/master/.github/workflows/dump_contexts.yaml

I could see this at the top of the logs:

```
Job defined at: HariSekhon/GitHub-Actions/.github/workflows/dump_contexts.yaml@refs/heads/master
```

but no other reference in the logs from all the dumped contexts that would be usable to find out the reusable workflow's repo in a portable programmatic way that I could reference in the reusable workflow itself (to source things from the reusable workflow's repo).

The use case for this would be as a workaround for issue #18601.

**Comment by MathiasReker** [comment]
You can use `needs`. Have a look at this example: https://github.com/MathiasReker/CI-CD/blob/develop/.github/workflows/ci-cd.yml

**Reply by HariSekhon** [reply]
`needs` only references another job in the same workflow, not the source repo of the imported reusable called workflow.

I need to find out programmatically what the reusable workflow repo is from the client's repo which is the caller and executing it.

**Comment by ajaykn** [comment]
@HariSekhon Currently we don't have context that has reusable workflow details (like called repo / called workflow path etc).

From docs: https://docs.github.com/en/actions/using-workflows/reusing-workflows
When a reusable workflow is triggered by a caller workflow, the github context is always associated with the caller workflow. 

Regarding the #18601 , please check the workaround https://github.com/orgs/community/discussions/18601#discussioncomment-4898425
Hope that helps for time being.

In future, may be we can have a context like `workflow` that can have details of called workflow.
 `workflow.repo` - repo that has the workflow.
 `workflow.sha` - sha of the workflow file that is being used

cc @h2floh, @avocaddo

**Reply by cmergenthaler** [reply]
accessing the called workflow's context would be really helpful in different scenarios

**Reply by salvianreynaldi** [reply]
like getting the composite action from the called workflow repository at the same commit sha https://github.com/orgs/community/discussions/18601#discussioncomment-8240123
